File: backend/catalogo/src/shared/core/Entity.js
export class Entity {
  constructor(id) {
    if (!id) throw new Error("Una entidad debe tener un ID válido.");
    this.id = id;
  }

  equals(other) {
    if (other == null || other === undefined) return false;
    if (this === other) return true;
    if (!(other instanceof Entity)) return false;
    return this.id === other.id;
  }
}

File: backend/pagos/payments/domain/shared/core/entity.py
class Entity:
    def __init__(self, entity_id):
        if not entity_id:
            raise ValueError("Una entidad debe tener un ID válido.")
        self._id = entity_id

    @property
    def id(self):
        return self._id

    def __eq__(self, other):
        if not isinstance(other, type(self)):
            return False
        return self._id == other.id

    def __hash__(self):
        return hash(self._id)